lib/utils/__tests__/parseSelectorAST.test.mjs

```diff
@@ -0,0 +1,48 @@
+import postcss from 'postcss';
+
+import parseSelectorAST from '../parseSelectorAST.mjs';
+
+describe('parseSelectorAST', () => {
+	it('parses selectors into an AST', async () => {
+		const result = await postcss().process('a .foo, b::before {}');
+		const rule = result.root.first;
+		const ast = parseSelectorAST(rule.selector, result, rule);
+
+		expect(ast).toBeTruthy();
+		expect(result.warnings()).toHaveLength(0);
+	});
+
+	it('short circuits on empty selectors', async () => {
+		const result = await postcss().process('{}');
+		const rule = result.root.first;
+		const ast = parseSelectorAST(rule.selector, result, rule);
+
+		expect(ast).toBeUndefined();
+		expect(result.warnings()).toHaveLength(0);
+	});
+
+	it('warns on invalid selectors, but does not throw', async () => {
```

Comment:
ditto

Reply:
This describes the public interface, not the implementation, so I am going to keep this as is.